examples/pony/passthrough/passthrough.pony

```diff
@@ -0,0 +1,275 @@
+              else
+                @printf[I32]("Unknown source type: %s\n".cstring(), st.cstring())
+                Fail()
+              end
+              if verbose then
+                @printf[I32]("source type: %s, parallelism = %lu\n".cstring(),
+                  st.cstring(), par_factor)
+              end
+            // There are lots of Wallaroo-specific flags that will
+            // be caught by a ParseError clause ... so omit that clause.
+            end
+          end
+          (src as Pipeline[InputBlob])
+        end
+
+        for o in recover ref make_options(env.args) end do
+          match o
+          | ("parallelism", let p: U64) =>
+            par_factor = p.usize()
+          | ("key-by", let kb: String) =>
+            match kb
+            | "first-byte" =>
+              source = source.key_by(FirstByte)
+            | "random" =>
+              source = source.key_by(KeyByRandomly)
+            else
+              @printf[I32]("Unknown key-by: %s\n".cstring(), kb.cstring())
+              Fail()
+            end
+            if verbose then
+              @printf[I32]("key by: %s\n".cstring(), kb.cstring())
+            end
+          | ("stage", let stg: String) =>
+            match stg
+            | "asis" =>
+              source = source.to[Array[U8] val](
+                AsIsC where parallelism = par_factor)
+            | "asis-state" =>
+              source = source.to[Array[U8] val](
+                AsIsStateC where parallelism = par_factor)
+            | "noop" =>
+              source = source.to[Array[U8] val](
+                NoOp where parallelism = par_factor)
+            else
+              @printf[I32]("Unknown stage: %s\n".cstring(), stg.cstring())
+              Fail()
+            end
+            if verbose then
+              @printf[I32]("stage: %s, parallelism = %lu\n".cstring(),
+                stg.cstring(), par_factor)
+            end
+          end
+        end
+
+        par_factor = 1 // Reset for typical default for sink
+        for o in recover ref make_options(env.args) end do
+          match o
+          | ("sink-parallelism", let p: U64) =>
+            par_factor = p.usize()
+          | ("sink", let sk: String) =>
+            if verbose then
+              @printf[I32]("sink parallelism: %lu\n".cstring(), par_factor)
+            end
+            match sk
+            | "tcp" =>
+              source = source
+                .to_sink(TCPSinkConfig[InputBlob].from_options(
+                  OutputBlobEncoder,
+                  TCPSinkConfigCLIParser(env.args)?(0)?)
+                  where parallelism = 1 /*** par_factor ***/)
+            | "connector" =>
+              source = source
+                .to_sink(ConnectorSinkConfig[InputBlob].from_options(
+                  OutputBlobEncoder,
+                  ConnectorSinkConfigCLIParser(env.args)?(0)?)
+                  where parallelism = par_factor)
```

Comment:
Same point here re: parsing order.  Even if currently parameters are parsed in the correct order, that depends on the implementation of the options library, so I think we should assume it could change.